lib/plugins/rotate_asg_instances/asg.rb

```diff
@@ -297,9 +300,17 @@ def shutdown_instance(id)
 
       def instance_in_terminal_state?(instance)
         state = if instance.is_a?(Aws::EC2::Instance)
-                  instance.state.name
+                  if instance.exists?
```

Comment:
Not sure if we can hit this is real-world testing, so maybe extending the unit test for this method to include one that simulates the response we think the AP returns in this edge case would be helpful.